### DISCUSSION TITLE
CGN Testing

### DISCUSSION THREAD

**Discussion by GIC-de** [rtbrick/bngblaster#54] [Ideas]
This conversation is about to collect requirements and ideas for CGN testing using BNG Blaster.

**Comment by GIC-de** [comment]
The general idea is to add support for TCP streams using [LwIP](https://en.wikipedia.org/wiki/LwIP) user space TCP with the target to support up to 1M TCP streams.

**Comment by ChrisDeh** [comment]
When thinking about CGN testing the first interesting things beside performance from my perpective would be:
* How are IP and Port Allocation per Subscriber handeled? (BPA (Bulk Port Allocation), Single IP per Subscriber, multiple IPs per Subscriber)
* Is BPA working properly?
* What happens when allowed Ports per User Limits are reached?
* Are things like ALG performed? (Cisco has default SIP ALG activated)

Hope these Ideas help!
